content/en/blog/2020/show-source-ip/index.md

```diff
@@ -0,0 +1,335 @@
+* ALPN, offering h2
+* ALPN, offering http/1.1
+* successfully set certificate verify locations:
+*   CAfile: new_certificates/example.com.crt
+  CApath: none
+* TLSv1.2 (OUT), TLS handshake, Client hello (1):
+* TLSv1.2 (IN), TLS handshake, Server hello (2):
+* TLSv1.2 (IN), TLS handshake, Certificate (11):
+* TLSv1.2 (IN), TLS handshake, Server key exchange (12):
+* TLSv1.2 (IN), TLS handshake, Server finished (14):
+* TLSv1.2 (OUT), TLS handshake, Client key exchange (16):
+* TLSv1.2 (OUT), TLS change cipher, Change cipher spec (1):
+* TLSv1.2 (OUT), TLS handshake, Finished (20):
+* TLSv1.2 (IN), TLS change cipher, Change cipher spec (1):
+* TLSv1.2 (IN), TLS handshake, Finished (20):
+* SSL connection using TLSv1.2 / ECDHE-RSA-CHACHA20-POLY1305
+* ALPN, server accepted to use h2
+* Server certificate:
+*  subject: CN=a25fa0b4835b.elb.us-west-2.amazonaws.com; O=httpbin organization
+*  start date: Oct 29 20:39:12 2020 GMT
+*  expire date: Oct 29 20:39:12 2021 GMT
+*  common name: a25fa0b4835b.elb.us-west-2.amazonaws.com (matched)
+*  issuer: O=example Inc.; CN=example.com
+*  SSL certificate verify ok.
+* Using HTTP2, server supports multi-use
+* Connection state changed (HTTP/2 confirmed)
+* Copying HTTP/2 data in stream buffer to connection buffer after upgrade: len=0
+* Using Stream ID: 1 (easy handle 0x7fbf8c808200)
+> GET /headers?show_env=1 HTTP/2
+> Host: a25fa0b4835b.elb.us-west-2.amazonaws.com
+> User-Agent: curl/7.64.1
+> Accept: */*
+>
+* Connection state changed (MAX_CONCURRENT_STREAMS == 2147483647)!
+< HTTP/2 200
+< server: istio-envoy
+< date: Thu, 29 Oct 2020 20:44:01 GMT
+< content-type: application/json
+< content-length: 612
+< access-control-allow-origin: *
+< access-control-allow-credentials: true
+< x-envoy-upstream-service-time: 1
+<
+{
+  "headers": {
+    "Accept": "*/*",
+    "Content-Length": "0",
+    "Host": "a25fa0b4835b.elb.us-west-2.amazonaws.com",
+    "User-Agent": "curl/7.64.1",
+    "X-B3-Sampled": "0",
+    "X-B3-Spanid": "69913a6e6e949334",
+    "X-B3-Traceid": "729d5da3618545da69913a6e6e949334",
+    "X-Envoy-Attempt-Count": "1",
+    "X-Envoy-Decorator-Operation": "httpbin.default.svc.cluster.local:8000/headers*",
+    "X-Envoy-Internal": "true",
+    "X-Forwarded-For": "172.16.5.30",
+    "X-Forwarded-Proto": "https",
+    "X-Request-Id": "299c7f8a-5f89-480a-82c9-028c76d45d84"
+  }
+}
+* Connection #0 to host a25fa0b4835b.elb.us-west-2.amazonaws.com left intact
+* Closing connection 0
+{{< /text >}}
+
+## Conclusion
+
+This blog presents the deployment of stack consisted of AWS NLB and Istio ingress gateway with enabling of proxy-protocol. We hope it is useful to you if you are interested at protocol enabling in an anecdotal, experiential, and more informal way. One thing needs your attention is X-Forwarded-For header used only for convenience of reading in test. Fake of X-Forwarded-For is a problem worthy to solve but not belong to the scope of this blog.
```

Comment:
```suggestion
This blog presents the deployment of a stack that consists of an AWS NLB and Istio ingress gateway that are enabled with proxy-protocol. We hope it is useful to you if you are interested in protocol enabling in an anecdotal, experiential, and more informal way. However, note that the `X-Forwarded-For` header should be used only for the convenience of reading in test, as dealing with fake `X-Forwarded-For` values is not within the scope of this blog.
```

I am not sure if my rewording of the last sentence is accurate. Please check

Reply:
Thanks. Done